File: Java/2138. Divide a String Into Groups of Size k.java
class Solution {
    public String[] divideString(String s, int k, char fill) {
        List<String> res = new ArrayList<>();
        StringBuilder sb = new StringBuilder();
        
        for (int i=0; i<s.length(); i++){
            sb.append(s.charAt(i));
            if (sb.length() == k){
                res.add(sb.toString());
                sb.setLength(0);
            }
        }
        if (sb.length()>0){
            while(sb.length()<k){
                sb.append(fill);
            }
            res.add(sb.toString());
        }
        return res.toArray(new String[0]);
    }
}

File: Python/2138. Divide a String Into Groups of Size k.py
class Solution:
    def divideString(self, s: str, k: int, fill: str) -> List[str]:
        res =[]
        substr = ""
        for i in range(len(s)):
            substr += s[i]
            print(substr)
            if len(substr) == k:
                res.append(substr)
                substr= ""
        if substr:
            substr += fill*(k-len(substr))
            res.append(substr)
        return res